src/analytic/GRD2D/GRD2DReconstruction.cxx

```diff
@@ -0,0 +1,518 @@
+			l1a = ceil((-u+PGx[ith][ip])*sp1dp); 
+			if(l1a<-sp1/2) l1a = -sp1/2; 
+			l1b = floor((u+PGx[ith][ip])*sp1dp); 
+			if(l1b>sp1/2-1) l1b = sp1/2-1; 
+			
+			l2a = ceil((-u+PGy[ith][ip])*sp1dp); 
+			if(l2a<-sp1/2) l2a = -sp1/2; 
+			l2b = floor((u+PGy[ith][ip])*sp1dp); 
+			if(l2b>sp1/2-1) l2b = sp1/2-1;
+			
+			for(int l1=l1a; l1<=l1b; l1++){ 
+				float T1 = l1/sp1dp-PGx[ith][ip];
+				float wKB1 = boost::math::cyl_bessel_i(0.0f,2*M_PI*ar*u*sqrt(1-pow(T1/u,2)))/(2.0f*u);
+				for(int l2=l2a; l2<=l2b; l2++){ 
+					float T2 = l2/sp1dp-PGy[ith][ip];
+					ff[l1+sp1/2][l2+sp1/2] = ff[l1+sp1/2][l2+sp1/2] + 
+						abs(ip-sp1/2.0f)*wKB1*((float)boost::math::cyl_bessel_i(0.0f,2*M_PI*ar*u*sqrt(1-pow(T2/u,2))))/(2.0f*u)*P[ith][ip];
+				}
+			}
+		}
+	}
+
+	
+	// gridding
+	ff = ff*dth/(sp1*dp);
+	
+	
+	fftshift(ff,sp1); 
+	inverse_fourier(ff);
+	fftshift(ff,sp1); 
+	
+	// gridding
+	int spf = 2*floor(sp/2); // make dimensions even
+	//float img[spf][spf]; 
+  std::vector<std::vector<float>> img(spf, std::vector<float>(spf)); // img := temporary reconstructed slice on spf×spf grid
+	for(int ix=-spf/2; ix<=spf/2-1; ix++){ 
+		for(int iy=-spf/2; iy<=spf/2-1; iy++){ 
+			img[ix+spf/2][iy+spf/2] = std::real(ff[ix+sp1/2][iy+sp1/2]/
+				((float)(sinh(2*M_PI*ar*u*sqrt(1-pow(ix*dp/ar,2)))/(2*M_PI*ar*u*sqrt(1-pow(ix*dp/ar,2)))*
+				 sinh(2*M_PI*ar*u*sqrt(1-pow(iy*dp/ar,2)))/(2*M_PI*ar*u*sqrt(1-pow(iy*dp/ar,2))))));
+		}
+	}
+
+	if(image.get_x_size() != sp || zoom != 1) { 
+		// perform bilinear interpolation 
+		if(iz==0) 
+		    info(stir::format("Image dimension mismatch: tangential positions {}, xy output {} — interpolating...",
+                  sp, image.get_x_size()));
+
+
+		//int sx = image.get_x_size(); 
+		//int sy = sx; // sx,sy := output image size (assumed square)
+  	    //std::vector<float> xn1(sx); 
+        //std::vector<float> yn1(sy); // xn1,yn1 := normalised target grids
+		//float dx1 = 2./(sx-1), dy1 = 2./(sy-1);  // dx1,dy1 := target grid steps
+		//float dx = 2./(sp-1), dy = 2./(sp-1); // dx,dy  := source grid steps
+		const int sx = image.get_x_size();
+        const int sy = image.get_y_size();
+        std::vector<float> xn1(sx), yn1(sy);
+        // mm -> normalised by detector half-FOV (requires arc-corrected data)
+        auto ac_sptr = dynamic_pointer_cast<const ProjDataInfoCylindricalArcCorr>(
+                     proj_data_ptr->get_proj_data_info_sptr());
+        if (!ac_sptr)
+           error("GRD2D: need arc-corrected proj data (ProjDataInfoCylindricalArcCorr).");
+        const float ts   = ac_sptr->get_tangential_sampling();
+        const float Rmax = 0.5f * ts * (sp - 1);
+       for (int ix = 0; ix < sx; ++ix) {
+          // get x_mm at voxel center; use suitable STIR helper to get physical coords
+        const float x_mm = image.get_physical_coordinates_for_indices(ix + image.get_min_x(),
+                                                                      image.get_min_y(), iz).x();
```

Comment:
this doesn't compile. you'd need to use `make_coordinate`. Moreover, STIR uses the opposite order, so this  would have to be

```suggestion
        const float x_mm = image.get_physical_coordinates_for_indices(iz,  image.get_min_y(), 
                                                                                                              ix + image.get_min_x()).x();
```